app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectFragment.kt

```diff
@@ -54,26 +66,45 @@ class DeviceConnectFragment : Fragment(), IDeviceConnectView {
     private var checkDevice: Boolean = false
     private val REQUEST_LOCATION_ACCESS = 101
     private val REQUEST_WIFI_ACCESS = 102
+    private lateinit var realm: Realm
+    private val availableRoomsList: ArrayList<AvailableRoomsFormat> = ArrayList()
+    private lateinit var availableRoomsRecyclerView: RecyclerView
+    private var availableRoomsAdapter: RecyclerView.Adapter<*>? = null
+    private lateinit var roomNameSelected: String
+    private lateinit var macId: String
+    private lateinit var rootView: View
 
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
         // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_device_connect, container, false)
+        rootView = inflater.inflate(R.layout.fragment_device_connect, container, false)
+        realm = Realm.getDefaultInstance()
+        availableRoomsRecyclerView = rootView.findViewById(R.id.rooms_available)
+        return rootView
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+
         mainWifi = context?.getSystemService(Context.WIFI_SERVICE) as WifiManager
         deviceConnectPresenter = DeviceConnectPresenter(requireContext(), mainWifi)
         deviceConnectPresenter.onAttach(this)
-        receiverWifi = WifiReceiver()
-        deviceConnectPresenter.searchDevices()
 
-        addDeviceButton.setOnClickListener {
-            deviceConnectPresenter.searchDevices()
+        val bundle = activity?.intent?.extras
+
+        if (bundle?.getString("roomName")?.isNullOrEmpty() == false) {
```

Comment:
I don't know why you still require to compare with false after a null check.

Reply:
Error is "Type mismatch: Inferred type is Boolean? but boolean was expected"